ironfish/src/mining/director.ts

```diff
@@ -503,22 +502,19 @@ export class MiningDirector {
     const nullifiers = new BufferSet()
 
     for (const transaction of this.memPool.get()) {
-      const conflicted = await AsyncUtils.find(transaction.spends(), async (spend) => {
-        if (nullifiers.has(spend.nullifier)) {
-          return true
-        }
-
-        if (await this.chain.nullifiers.contains(spend.nullifier)) {
-          return true
-        }
-
-        return false
+      const conflicted = await AsyncUtils.find(transaction.spends(), (spend) => {
+        return Promise.resolve(nullifiers.has(spend.nullifier))
```

Comment:
I wonder if I can update the types here in another PR so that the async version can at least take a non async method.